packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -1499,15 +1523,41 @@ export class BaseResolversVisitor<
         parentType,
         parentTypeSignature: this.getParentTypeForSignature(node),
       });
-      const mappedTypeKey = isSubscriptionType ? `${mappedType}, "${node.name}"` : mappedType;
 
-      const directiveMappings =
-        node.directives
-          ?.map(directive => this._directiveResolverMappings[directive.name as any])
-          .filter(Boolean)
-          .reverse() ?? [];
+      const { mappedTypeKey, resolverType } = ((): { mappedTypeKey: string; resolverType: string } => {
+        const baseType = getBaseTypeNode(original.type);
+        const realType = baseType.name.value;
+        const typeToUse = this.getTypeToUse(realType);
+        /**
+         * Turns GraphQL type to TypeScript types (`mappedType`) e.g.
+         * - String!  -> ResolversTypes['String']>
+         * - String   -> Maybe<ResolversTypes['String']>
+         * - [String] -> Maybe<Array<Maybe<ResolversTypes['String']>>>
+         * - [String!]! -> Array<ResolversTypes['String']>
+         */
+        const mappedType = this._variablesTransformer.wrapAstTypeWithModifiers(typeToUse, original.type);
+
+        const subscriptionType = this._schema.getSubscriptionType();
+        const isSubscriptionType = subscriptionType && subscriptionType.name === parentName;
+
+        if (isSubscriptionType) {
+          return {
+            mappedTypeKey: `${mappedType}, "${node.name}"`,
+            resolverType: 'SubscriptionResolver',
+          };
+        }
+
+        const directiveMappings =
+          node.directives
+            ?.map(directive => this._directiveResolverMappings[directive.name as any])
+            .filter(Boolean)
+            .reverse() ?? [];
 
-      const resolverType = isSubscriptionType ? 'SubscriptionResolver' : directiveMappings[0] ?? 'Resolver';
+        return {
+          mappedTypeKey: this.modifyFieldDefinitionNodeTransformedType({ node: original, originalType: mappedType }),
+          resolverType: directiveMappings[0] ?? 'Resolver',
+        };
+      })();
```

Comment:
I refactored this bit to bring related variables together, in the same scope - so it's easier to read